src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/GeneratedVirtualDocument.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.AspNetCore.Razor.Telemetry;
+using Microsoft.VisualStudio.LanguageServer.ContainedLanguage;
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor;
+
+internal abstract class GeneratedVirtualDocument<T>(Uri uri, ITextBuffer textBuffer, ITelemetryReporter? telemetryReporter) : VirtualDocumentBase<T>(uri, textBuffer) where T : VirtualDocumentSnapshot
+{
+    private readonly ITelemetryReporter? _telemetryReporter = telemetryReporter;
```

Comment:
Does this need to be nullable? Or, do we always have an ITelemetryReporter to pass here in production code?

Reply:
Good spot, will check, I suspect now this might just be for tests so I can fix them up.